packages/react/slot/src/slot.test.tsx

```diff
@@ -160,3 +191,24 @@ const Button = React.forwardRef<
     </Comp>
   );
 });
+
+const Input = React.forwardRef<
+  React.ElementRef<'input'>,
+  React.ComponentProps<'input'> & {
+    asChild?: boolean;
+  }
+>(({ asChild, children, ...props }, forwardedRef) => {
+  const Comp = asChild ? Slot : 'input';
+  const [value, setValue] = useState('');
```

Comment:
using controlled value here to force component to re-render as value is typed in the input